core/common/src/main/java/alluxio/util/io/PathUtils.java

```diff
@@ -187,7 +189,12 @@ public static void validatePath(String path) throws InvalidPathException {
    * @return a deterministic temporary file name
    */
   public static String temporaryFileName(long nonce, String path) {
-    return path + ".alluxio." + String.format("0x%16X", nonce) + ".tmp";
+    try {
+      return path + ".alluxio." + URLEncoder.encode(String.format("0x%16X", nonce),
+          "UTF-8") + ".tmp";
+    } catch (UnsupportedEncodingException e) {
+      return path + ".alluxio." + String.format("0x%16X", nonce) + ".tmp";
+    }
```

Comment:
Do we expect to hit unsupported encoding exceptions? A different work around may be to avoid getting spaces, but given the current way we generate the nonce it may not be simple.